src/components/VaccinationsTable/utils.ts

```diff
@@ -1,53 +1,68 @@
-import regions, { State } from 'common/regions';
-import sortBy from 'lodash/sortBy';
+import regions, { Region } from 'common/regions';
+import orderBy from 'lodash/orderBy';
 import take from 'lodash/take';
 import takeRight from 'lodash/takeRight';
+import { MapView } from 'screens/HomePage/HomePage';
 import { LocationSummariesByFIPS } from 'common/location_summaries';
+import { Metric } from 'common/metricEnum';
 
-export interface StateWithVaccinationMetrics {
-  stateName: string;
-  vaccinationsInitiated: number | null;
-  vaccinationsCompleted: number | null;
-}
-
-export interface StateWithVaccinationMetricsAndRank {
+export interface RegionVaccinationInfo {
+  regionName: string;
+  vaccinationsInitiated: number;
+  vaccinationsCompleted: number;
   rank: number;
 }
 
-function getStatesSortedByVaccinationsCompleted(): StateWithVaccinationMetricsAndRank[] {
-  const statesWithVaccinationMetrics = regions.states.map(
-    (state: State, i: number) => {
-      const summaryForFips = LocationSummariesByFIPS[state.fipsCode];
-      return {
-        stateName: state.name,
-        vaccinationsInitiated: summaryForFips.metrics[6]?.value ?? null,
-        vaccinationsCompleted: summaryForFips.vc ?? null,
-      };
-    },
+function getRegionsSortedByVaccinationsInitiated(
+  regionScope: MapView,
+): RegionVaccinationInfo[] {
+  const regionsToSort =
+    regionScope === MapView.COUNTIES ? regions.counties : regions.states;
+
+  const regionsWithVaccinationMetrics = regionsToSort.map((region: Region) => {
+    const summaryForFips = LocationSummariesByFIPS[region.fipsCode];
+    return {
+      regionName: region.name,
+      vaccinationsInitiated:
+        summaryForFips?.metrics[Metric.VACCINATIONS]?.value ?? null,
+      vaccinationsCompleted: summaryForFips?.vc ?? null,
+    };
+  });
+
+  const sortedByVaccinationsInitiated = orderBy(
+    regionsWithVaccinationMetrics,
+    region => region.vaccinationsInitiated,
+    'desc',
   );
-  const sorted = sortBy(
-    statesWithVaccinationMetrics,
-    (state: StateWithVaccinationMetrics) => state.vaccinationsInitiated,
-  ).reverse(); // reversing because sortBy sorts in ascending order, placing lowest-ranked first
-  const sortedWithRank = sorted.map(
-    (stateInfo: StateWithVaccinationMetrics, i: number) => ({
-      stateInfo,
-      rank: i + 1,
-    }),
+
+  const sortedNoNulls = sortedByVaccinationsInitiated.filter(
+    region =>
+      Number.isFinite(region.vaccinationsInitiated) &&
+      Number.isFinite(region.vaccinationsCompleted),
   );
+
+  const sortedWithRank = sortedNoNulls.map((regionInfo, i: number) => ({
+    rank: i + 1,
+    regionName: regionInfo.regionName,
+    vaccinationsInitiated: regionInfo.vaccinationsInitiated as number,
+    vaccinationsCompleted: regionInfo.vaccinationsCompleted as number,
```

Comment:
nit: Rather than `as number` here I would add `as RegionVaccinationInfo[]` on line 44 after the `filter()`. That way the cast is directly tied to the code that we are relying on to remove the nulls (since any time you're casting you could accidentally shoot yourself in the foot if it turns out you were wrong about what the type will be at runtime 😁 ).